peepingtom/utils/generic.py

```diff
@@ -12,31 +10,54 @@ def listify(obj):
     """
     # avoid circular import
     from ..datablocks import DataBlock
-    if isinstance(obj, Iterable) and not isinstance(obj, (str, DispatchList, DataBlock, Path)):
-        return list(obj)
+    if isinstance(obj, Iterable):
+        if isinstance(obj, (str, DataBlock, Path)):
+            return [obj]
+        else:
+            return list(obj)
     if obj is None:
         return []
     return [obj]
 
 
-def wrapper_method(other_func, ignore_args=0):
+def inherit_signature(*other_functions, ignore_args=None, add_args={}):
```

Comment:
love this

Reply:
should adding be in a separate decorator? feels like a different task

Reply:
Ignoring has to be there to avoid name clashes and stuff... so I figured adding might as well be in the same place :P Anyways it's a bit of a test, we'll see if this is usable or ends up being more work than anything.